fv3gfs/physics/physics_state.py

```diff
@@ -1,75 +1,134 @@
-        ArgSpec("w", "vertical_wind", "m/s", intent="inout"),
-        ArgSpec("omga", "vertical_pressure_velocity", "Pa/s", intent="inout"),
-    )
+    """
+    Physics state variables
+    From dynamical core:
+        qvapor: specific_humidity
+        qliquid: cloud_water_mixing_ratio
+        qrain: rain_mixing_ratio
+        qsnow: snow_mixing_ratio
+        qice: cloud_ice_mixing_ratio
+        qgraupel: graupel_mixing_ratio
+        qo3mr: ozone_mixing_ratio
+        qsgs_tke: turbulent_kinetic_energy
+        qcld: cloud_fraction
+        pt: air_temperature
+        delp: pressure_thickness_of_atmospheric_layer
+        delz: vertical_thickness_of_atmospheric_layer
+        ua: eastward_wind
+        va: northward_wind
+        w: vertical_wind
+        omga: vertical_pressure_velocity
+    Physics driver:
+        delprsi: model_level_pressure_thickness_in_physics
+        phii: interface_geopotential_height
+        phil: layer_geopotential_height
+        dz: geopotential_height_thickness
+        *_t1: respective dynamical core variables marched by 1 time step
+    Microphysics:
+        wmp: layer_mean_vertical_velocity_microph
+    """
 
-    def __init__(self, state, grid):
-        self.physics_state = self.make_physics_state(state)
-        self.physics_state = get_namespace(self.arg_specs, self.physics_state)
-        self.grid = grid
+    qvapor: FloatField
+    qliquid: FloatField
+    qrain: FloatField
+    qsnow: FloatField
+    qice: FloatField
+    qgraupel: FloatField
+    qo3mr: FloatField
+    qsgs_tke: FloatField
+    qcld: FloatField
+    pt: FloatField
+    delp: FloatField
+    delz: FloatField
+    ua: FloatField
+    va: FloatField
+    w: FloatField
+    omga: FloatField
+    delprsi: FloatField
+    phii: FloatField
+    phil: FloatField
+    dz: FloatField
+    wmp: FloatField
+    qvapor_t1: FloatField
+    qliquid_t1: FloatField
+    qrain_t1: FloatField
+    qsnow_t1: FloatField
+    qice_t1: FloatField
+    qgraupel_t1: FloatField
+    qcld_t1: FloatField
+    pt_t1: FloatField
+    ua_t1: FloatField
+    va_t1: FloatField
 
-    def make_physics_state(self, dycore_state):
-        phy_vars = [
-            "air_temperature",
-            "specific_humidity",
-            "cloud_water_mixing_ratio",
-            "rain_mixing_ratio",
-            "cloud_ice_mixing_ratio",
-            "snow_mixing_ratio",
-            "graupel_mixing_ratio",
-            "ozone_mixing_ratio",
-            "turbulent_kinetic_energy",
-            "cloud_fraction",
-            "eastward_wind",
-            "northward_wind",
-            "vertical_wind",
-            "pressure_thickness_of_atmospheric_layer",
-            "vertical_thickness_of_atmospheric_layer",
-            "vertical_pressure_velocity",
-        ]
-        phy_state = {key: copy.deepcopy(dycore_state[key]) for key in phy_vars}
-        return phy_state
+    @classmethod
+    def from_dycore_state(cls, state, storage: FloatField) -> "PhysicsState":
```

Comment:
Ideally, the physics and dycore states shouldn't know about each other. This can be done by writing a MainLoopState which has properties `dynamics` and `physics` (or similar names), and constructs both of these using the same storages where values are overlapping.

Reply:
What about stencils shared between the dycore and physics? For example, `moist_cv` and `CubedToLatLon`.

Reply:
You can pass relevant states in as independent variable and have them called through